### PR TITLE
tests: improvements to test coverage

### PR DESCRIPTION
**Pull request by SamiHiltunen**
* Adds more tests to increase the coverage of tests.

Signed-off-by: Sami Hiltunen <sami.mikael.hiltunen@cern.ch>

### CODE REVIEW

File: invenio_oauth2server/ext.py [hunk 1]
```diff
@@ -118,7 +117,12 @@ def init_config(self, app):
             'OAUTH2SERVER_SETTINGS_TEMPLATE',
             app.config.get('BASE_SETTINGS_TEMPLATE',
                            'invenio_oauth2server/settings/base.html'))
+        app.config.setdefault('OAUTH2_CACHE_TYPE', 'simple')
```
Comment (jirikuncar) [inline invenio_oauth2server/ext.py]: @SamiHiltunen revert this changes. everything is already set in https://github.com/inveniosoftware/invenio-oauth2server/blob/master/invenio_oauth2server/config.py

File: invenio_oauth2server/ext.py [hunk 2]
```diff
@@ -118,7 +117,12 @@ def init_config(self, app):
...
+        app.config.setdefault('OAUTH2_CLIENT_ID_SALT_LEN', 40)
+        app.config.setdefault('OAUTH2_CLIENT_SECRET_SALT_LEN', 60)
+        app.config.setdefault('OAUTH2_TOKEN_PERSONAL_SALT_LEN', 60)
+        app.config.setdefault('OAUTH2_PROVIDER_TOKEN_EXPIRES_IN', 3600)
 
-        for k in dir(config):
+        for k in dir(app.config):
```
Comment (jirikuncar) [inline invenio_oauth2server/ext.py]: `dir(config)` is correct

File: invenio_oauth2server/views/server.py [hunk 3]
```diff
@@ -161,14 +162,3 @@ def info():
         ))
     else:
         abort(404)
-
-
-@blueprint.route('/invalid/')
```
Comment (jirikuncar) [inline invenio_oauth2server/views/server.py]: IMHO you should not remove it!

File: invenio_oauth2server/config.py [hunk 4]
```diff
@@ -24,7 +24,7 @@
 
 """OAuth2Server configuration variables."""
 
-OAUTH2_CACHE_TYPE = 'redis'
```
Comment (jirikuncar) [inline invenio_oauth2server/config.py]: why do you need to change it? redis seems like reasonable default choice.

File: invenio_oauth2server/views/settings.py [hunk 5]
```diff
@@ -167,14 +167,14 @@ def client_new():
 def client_view(client):
     """Show client's detail."""
     if request.method == 'POST' and 'delete' in request.form:
-        db.session.delete(client)
-        db.session.commit()
+        with db.session.begin_nested():
```
Comment (jirikuncar) [inline invenio_oauth2server/views/settings.py]: using `begin_nested` should not be necessary in views

File: invenio_oauth2server/config.py (hunk 4 above)
Comment (SamiHiltunen) [inline invenio_oauth2server/config.py]: I figured it should be 'simple' as redis is an optional dependency. I've changed it back to redis as a default.
